platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ConfigurationFilesPoller.java

```diff
@@ -58,100 +54,65 @@
         try {
-            WatchKey key;
-
-            while (!Thread.currentThread()
-                    .isInterrupted()) {
-                key = watchService.take(); // blocking
-                LOGGER.debug("Key has been signalled.  Looping over events.");
-
-                for (WatchEvent<?> genericEvent : key.pollEvents()) {
-                    WatchEvent.Kind<?> kind = genericEvent.kind();
-
-                    if (kind == OVERFLOW || kind == ENTRY_MODIFY || kind == ENTRY_DELETE) {
-                        LOGGER.debug("Skipping event [{}]", kind);
-                        continue;
-                    }
-
-                    Path filename = (Path) genericEvent.context();
-
-                    if (!filename.toString()
-                            .endsWith(fileExtension)) {
-                        LOGGER.debug(
-                                "Skipping event for [{}] due to unsupported file extension of [{}].",
-                                filename,
-                                fileExtension);
-                        continue; // just skip to the next event
-                    }
-
-                    if (changeListener != null) {
-                        // Sleeping before notifying the listener to make sure file is
-                        // done writing, otherwise the listener may read the file too soon.
-                        TimeUnit.SECONDS.sleep(1);
-                        LOGGER.debug("Notifying [{}] of event [{}] for file [{}].",
-                                changeListener.getClass()
-                                        .getName(),
-                                kind,
-                                configurationDirectoryPath.resolve(filename));
-                        changeListener.notify(configurationDirectoryPath.resolve(filename));
-                    }
-                }
-
-                // Reset key, shutdown watcher if directory not able to be observed
-                // (possibly deleted)
-                if (!key.reset()) {
-                    LOGGER.warn("Configurations in [{}] are no longer able to be observed.",
-                            configurationDirectoryPath.toString());
-                    break;
-                }
+            String filename = file.getName();
+            LOGGER.debug("Watcher has been notified. Handling event for [{}].", filename);
+
+            if (changeListener != null) {
+                waitForFileToBeCompletelyWritten(file);
+                LOGGER.debug("Notifying [{}] of creation for file [{}].",
+                        changeListener.getClass()
+                                .getName(),
+                        file.getCanonicalPath());
+                changeListener.notify(Paths.get(file.getCanonicalPath()));
             }
-        } catch (InterruptedException | RuntimeException e) {
-            LOGGER.debug("The [{}] was interrupted.",
+        } catch (InterruptedException | IOException e) {
```

Comment:
❓Wonder if we should split this catch into `InterruptedException` and `IOException`. In the former we could do what we do now, but in the latter we should be able to log something and keep going. We should also consider what to do with `RuntimeException`. Thoughts?